sdk/provisioning/Azure.Provisioning.CloudMachine/src/CloudMachineInfrastructure.cs

```diff
@@ -0,0 +1,216 @@
+using Azure.Provisioning.ServiceBus;
+using Azure.Provisioning.Storage;
+
+namespace Azure.Provisioning.CloudMachine;
+
+public class CloudMachineInfrastructure : Infrastructure
+{
+    private readonly string _name;
+    private UserAssignedIdentity _identity;
+    private StorageAccount _storage;
+    private BlobService _blobs;
+    private BlobContainer _container;
+    private ServiceBusNamespace _serviceBusNamespace;
+    private ServiceBusNamespaceAuthorizationRule _serviceBusNamespaceAuthorizationRule;
+    private ServiceBusTopic _serviceBusTopic_main;
+    private ServiceBusTopic _serviceBusTopic_app;
+    private ServiceBusSubscription _serviceBusSubscription_main;
+    private ServiceBusSubscription _serviceBusSubscription_app;
+    private SystemTopic _eventGridTopic_Blobs;
+    private SystemTopicEventSubscription _systemTopicEventSubscription;
+
+    /// <summary>
+    /// The common principalId parameter.
+    /// </summary>
+    public BicepParameter PrincipalIdParameter => new BicepParameter("principalId", typeof(string));
+
+    /// <summary>
+    /// The common principalType parameter.
+    /// </summary>
+    public BicepParameter PrincipalTypeParameter => new BicepParameter("principalType", typeof(string));
+
+    /// <summary>
+    /// The common principalName parameter.
+    /// </summary>
+    public BicepParameter PrincipalNameParameter => new BicepParameter("principalName", typeof(string));
+
+    public CloudMachineInfrastructure(string? name = "cm") : base(name!)
+    {
+        _name = name ?? "cm";
+        _identity = new($"{_name}_identity");
+        var managedServiceIdentity = new ManagedServiceIdentity
+        {
+            ManagedServiceIdentityType = ManagedServiceIdentityType.UserAssigned,
+            UserAssignedIdentities = { { BicepFunction.Interpolate($"{_identity!.Id}").Compile().ToString(), new UserAssignedIdentityDetails() } }
+        };
+
+        _storage = StorageResources.CreateAccount($"{_name}_sa");
+        _storage.Identity = managedServiceIdentity;
+
+        _blobs = new($"{_name}_blobs")
+        {
+            Parent = _storage,
+        };
+        _container = new BlobContainer($"{_name}_container", "2023-01-01")
+        {
+            Parent = _blobs,
+            Name = "default"
+        };
+
+        _serviceBusNamespace = new($"{_name}_sb")
+        {
+            Sku = new ServiceBusSku
+            {
+                Name = ServiceBusSkuName.Standard,
+                Tier = ServiceBusSkuTier.Standard
+            },
+        };
+        _serviceBusNamespaceAuthorizationRule = new($"{_name}_sb_auth_rule", "2021-11-01")
+        {
+            Parent = _serviceBusNamespace,
+            Rights = [ServiceBusAccessRight.Listen, ServiceBusAccessRight.Send, ServiceBusAccessRight.Manage]
+        };
+        _serviceBusTopic_main = new($"{_name}_sb_topic_main", "2021-11-01")
+        {
+            Parent = _serviceBusNamespace,
+            // Name = "default",
```

Comment:
Why is this commented out?  I don't know if I'm aware of whatever's blocking this.

Reply:
I don't think it was intentional.